lib/mirage/mirage.mli

```diff
@@ -764,6 +764,104 @@ val default_argv : argv impl
 val no_argv : argv impl
 (** [no_argv] Disable command line parsing and set argv to [|""|]. *)
 
+(** {2 Mimic device} *)
+
+(** A {i mimic} device is a {i context} which contains few informations to let
+    the user to [connect]/{!Mimic.resolve}/allocate a new resource to
+    communicate with a peer. Depending on what the context contains, the
+    protocol used to communicate can be TCP/IP, TLS or SSH (or something else).
```

Comment:
It has to be network-based though? It can't be, for example, Vchan?

Reply:
It can be `vchan` 👍 . As far as the protocol provides a `Mirage_flow.S`, it can be used as a _transmission protocol_ for the Smart protocol.